pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -272,6 +272,11 @@ func resolveDataWithTemplateVars(ctx context.Context, data integration.Data, svc
 			if err != nil {
 				return data, err
 			}
+			// If a `‰` character hasn’t been consumed by a `%%var%%` template variable replacement,
+			// let’s restore it to the initial `%%` value it had in the original string.
+			if str, ok := s.(string); ok {
+				s = strings.ReplaceAll(str, "‰", "%%")
+			}
```

Comment:
Even though it's quite unlikely, on the off chance that someone actually uses the `‰` character in a config, this solution would still have the surprising behavior of replacing that character with `%%`.

Admittedly a solution that would avoid that entirely (for example something like choosing the delimiter dynamically to avoid clashing with anything that's actually present in the input) might be too involved just to take care of such an edge case, so I wonder whether it's worth the effort.